### ISSUE TITLE
Fuzzy search

### ISSUE THREAD

**Issue by allochi** [karmi/retire#172]
Hi,

First, thanks for the wonderful gem, I love it, and I spent the whole day exploring around, I'm intending to use it with my projects one I find out how to implement all the functionalities I need.

Now, I tried to use the fuzzy search in active-record, but I don't know how, maybe I'm new and I need to spend more time with the gem/elastic-search. 

My work around was to add a fuzzy method to query like this:

class Tire::Search::Query
  def fuzzy(match={})
    @value = { :fuzzy => match  }
  end
end

Could you please explain how to avoid this? or not, and add this to the Query class :D

Thanks again.

**Comment by karmi** [comment]
Hi, you can do fuzzy searches with the normal [Lucene query syntax](http://lucene.apache.org/java/3_5_0/queryparsersyntax.html#Fuzzy%20Searches): `query { string "sour~" }`.

You can also pass Hash/JSON to _Tire_, search the Readme for "fuzzy". (And, preferably, _do_ read it! :)

Of course, it could be added as a dedicated method to `Tire::Search::Query`, when includes unit tests etc.

**Comment by allochi** [comment]
Many thanks Karmi! I went through the whole read me yesterday, and tested the code, I must say that I don't have that much experience with Lucene, but this going to be this weekend project.

One last note about the read me, in facets, this:

facet 'global-tags' do
    terms :tags, :global => true
end

needs to change to:

facet 'global-tags', :global => true do
    terms :tags
end

global for facet not terms, Kimchy helped me finding this out. 

Also I found out that for this example to work, and return the right figures, we need take care of the sort by providing "not analyzed" title, we need to map title like this.

:title => {
   :type => 'multi_field', :fields => {
      :title => { :type => "string", :analyzer => "snowball" },
      :untouched => { :type => "string", :index => :not_analyzed }
}

and sorting to change to:

sort { by "title.untouched", 'desc' }

This is not to be taken as me correcting the great job that you are doing, I just hope to ease the startup of new comers like me as a thankful user of Tire.

Again, Thanks!
